### PR TITLE
Add @shopify/react-web-worker

### PR DESCRIPTION
**Pull request by ismail-syed**
Pretty small package here. 

* Provides a `useWorker` hook
* Re-exports all of `@shopify/web-worker`

### CODE REVIEW

File: packages/react-web-worker/src/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import {useLazyRef} from '@shopify/react-hooks';
+
+export function useWorker<T>(creator: () => T) {
+  const {current: worker} = useLazyRef(creator);
+
+  useEffect(
+    () => {
+      return () => {
+        // terminate(worker);
```
Comment (ismail-syed) [inline packages/react-web-worker/src/hooks.ts]: Will add in once #1111 is merged.
I'm ok with shipping this independently from #1111

File: packages/react-web-worker/src/test/hooks.test.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  return () => {
+    return {
+      hello(friend: string) {
+        return `Hello, ${friend}`;
+      },
+    };
+  };
+}
+describe('useWorker', () => {
+  function MockComponent({create = mockCreate(), children}) {
```
Comment (lemonmade) [inline packages/react-web-worker/src/test/hooks.test.tsx]: We should need to provide a type here (IIRC, TS was complaining in the editor about this prop). Yes, you will find that quilt's CI does not check types for tests, which is fairly shocking, I agree :P

File: packages/react-web-worker/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+```tsx
+import React, {useEffect} from 'react';
+import {Page} from '@shopify/polaris';
+import {createWorker, useWorker} from '@shopify/react-web-worker';
+
+// assume ./worker.ts contains
+// export function hello(name) {
+//  return `Hello, ${name}`;
+// }
+const create = createWorker(() => import('./worker'));
```
Comment (lemonmade) [inline packages/react-web-worker/README.md]: Newline above

File: packages/react-web-worker/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      })();
+    },
+    [worker],
+  );
+
+  return <Page title="Home"> {message} </Page>;
+}
+```
+
+### `@shopify/web-worker`
```
[the remark this answers is not in the document]
Comment (ismail-syed) [reply packages/react-web-worker/README.md]: Moving to the top

**Pull request merged**